health-checker/src/main/java/io/stargate/health/StorageHealthChecker.java

```diff
@@ -0,0 +1,158 @@
+          .build();
+  private final DataStoreFactory dataStoreFactory;
+
+  public StorageHealthChecker(DataStoreFactory dataStoreFactory)
+      throws ExecutionException, InterruptedException {
+
+    this.dataStoreFactory = dataStoreFactory;
+    if (SHOULD_CREATE_KS_AND_TABLE) {
+      ensureTableExists(dataStoreFactory.createInternal());
+    }
+  }
+
+  @Override
+  protected Result check() throws Exception {
+    if (!STORAGE_CHECK_ENABLED) {
+      return Result.healthy("Storage check disabled");
+    }
+    try {
+      DataStore dataStore = dataStoreFactory.createInternal();
+
+      Instant writeTimestamp = Instant.now();
+      // insert record
+      dataStore
+          .execute(
+              dataStore
+                  .queryBuilder()
+                  .insertInto(KEYSPACE_NAME, TABLE_NAME)
+                  .value(PK_COLUMN_NAME, STARGATE_NODE_ID)
+                  .value(VALUE_COLUMN_NAME, VALUE_COLUMN_VALUE)
+                  .ttl(INSERT_TTL_SECONDS)
+                  .timestamp(writeTimestamp.toEpochMilli())
+                  .build()
+                  .bind())
+          .get();
+
+      // select record
+      ResultSet rs =
+          dataStore
+              .execute(
+                  dataStore
+                      .queryBuilder()
+                      .select()
+                      .writeTimeColumn(VALUE_COLUMN_NAME)
+                      .from(KEYSPACE_NAME, TABLE_NAME)
+                      .where(PK_COLUMN_NAME, Predicate.EQ, STARGATE_NODE_ID)
+                      .build()
+                      .bind())
+              .get();
+
+      Row row = rs.one();
+      Instant timestampRead = Instant.ofEpochMilli(row.getLong(0));
+      if (isGreaterThanOrEqual(timestampRead, writeTimestamp)) {
+        return Result.healthy("Storage is operational");
+      } else {
+        return Result.unhealthy("Storage did not return the proper data.");
+      }
+    } catch (Exception e) {
+      logger.warn("checkIsReady failed with {}", e.getMessage(), e);
+      return Result.unhealthy("Unable to access Storage: " + e);
+    }
+  }
+
+  private boolean isGreaterThanOrEqual(Instant timestampRead, Instant expectedTimestamp) {
+    if (timestampRead == null) {
+      return false;
+    }
+    // in case there was a concurrent update, validate that the timestampRead is greater than
+    // expected timestamp
+    return expectedTimestamp.equals(timestampRead) || timestampRead.isAfter(expectedTimestamp);
+  }
+
+  private void ensureTableExists(DataStore dataStore)
+      throws ExecutionException, InterruptedException {
+    // create a dedicated health-check keyspace to not risk conflicts with users keyspace
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .keyspace(KEYSPACE_NAME)
+                .ifNotExists()
+                .withReplication(Replication.simpleStrategy(1))
```

Comment:
Sorry, I did not notice this before. In a replicated env. with `RF==1` shutting down any storage node will make _all_ Stargate nodes "unhealthy"... Would it make sense to make this parameter configurable so that people would not have to always use custom health checker DDL in realistic deployments?

Reply:
As far as I can see, the same RF is used for `auth` and `grapqhQL` keyspaces. 
If we want to make it configurable, shouldn't we do the same for other keyspaces?

Reply:
auth has to be custom-created in prod. I was just thinking that health-checking can avoid that, but perhaps is has to use custom DDL too :thinking: 

What is  the `graphQL` keyspace that you refer to? Where is it created? I do not know about that off the top of my head.

Reply:
GraphQL:
https://github.com/stargate/stargate/pull/634/files#diff-e181be50d5f73de330179580432cbe71c8b32630dcbb05a81af75bbd7ab88fa4R196

Reply:
Well, I think GraphQL has a similar issue in this case - added a comment on that PR.

Reply:
I can make it configurable, I am only afraid that this Health-Check already has a lot of config settings.

Reply:
True, but I believe we have use cases for all of them :)

For RF, in particular, it really depends on the deployment, and `1` is not a viable option in production... I do not really see how we can have a default for that that would work in all cases. 

Since the checker is on by default, this PR might introduce unexpected behaviour changes into environments where people already use Stargate and have replicated storage.

The alternative, I guess, would be to avoid running DDL at all, and have this checker disabled by default. Users would have to define the checker's table as part of their workflows and then configure it to use the custom table.

Reply:
Auth needs to be more configurable

https://github.com/stargate/stargate/issues/739

Reply:
I am yielding towards disabling it by default and enabling in classic deployment (Non-CNDB). It will be easier to use for our clients (CNDB,...) if it runs with just defaults. For other cases, we may enable it explicitly.
Implemented the RF setting in https://github.com/stargate/stargate/pull/742/commits/c968175c6d6736dad30851bd2503ac0770f1c38b

Reply:
SGTM. @dougwettlaufer : WDYT?

Reply:
Yeah I think that makes sense. It's a heavier weight check so "opt-in" is probably a better solution.